Project review by mentor

nzediegwu1 opened the pull request:
#### What does this PR do?
Project review by mentor

#### Description of Task to be completed?
Review of the whole code base of eventmanager web applicationh

#### How should this be manually tested?
npm run start
npm run client

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories? (If applicable)
Not yet

#### Screenshots (If applicable)
None

Commit:
Merge branch 'develop' of https://github.com/nzediegwu1/EventManager.git

Commit:
correct swagger controller path to use transpiled js

Commit:
Write more working unit tests for routes

*Include secret_key in travis.yml environment variable
*Remove custom eslint rules

nzediegwu1 committed:
Merge pull request #46 from nzediegwu1/ch/unit-tests

Write more working unit tests for routes

Commit:
Include copy-files command in npm build script

*To eliminate the need for babel-import module used for swagger json file import in server.js, which dragged down coverage report

Commit:
Include copy-files command omitted

nzediegwu1 committed:
update coveralls badge in readme

nzediegwu1 committed:
Update coveralls badge in readme

Commit:
Setup webpack

*Install dependencies for webpack
*Install babel loader and configure babel for react
*Create and configure webpack.config
*Install and configure react-hot-loader for HMR
*Create app.js, helloComponent.js in Client/src for testing babel setup for react
*Create index.html and import bundle.js in Client/dist folder

Commit:
Correct errors in HMR implementation

*Changed entry.public path to '/' in webpack.config.js
*Moved react-hot-loader/patch to the top of webpack entry
*Included AppContainer in app.js

nzediegwu1 committed:
Merge pull request #47 from nzediegwu1/ft/web-pack-integration-testing

Setup webpack for react

Commit:
Remove unnecessary code

Commit:
Create signinComponent

*Install dependencies for css, file-upload and style-loading
*Create signinComponent.js
*Create recoverPassword component
*Include resources for template into client folder and link to app

Commit:
*Install bootstrap, popper.js and jquery
*Configure webpack for bootstrap
*Refractor app.js
*Create bodyComponent for combining other components
*Create formGroup component for reusability of bootstrap form group
*Implement formgroup component in signin and recovery password components
*Delete helloComponent, bootstrap.min.js, jquery.min.js and popper.min.js files
*Tweak custom.css to correct design issues

nzediegwu1 committed:
Merge pull request #48 from nzediegwu1/ft/signin-react-component

Implement signinComponent

Commit:
1. Create signupComponent
2. Implement alternating display/hide of signin and signup forms using jquery: commented out
3. Define changeState method in signinComponent for alternating display/hide of signin and signup forms on clicking element with className: welcome

nzediegwu1 committed:
Merge pull request #49 from nzediegwu1/ft/react-signup-component

Implement SignupComponent

Commit:
Implement react dashboard component

1. Create navbarComponent in react using navbar template
2. Create sidebar component using HTML template
3. Create dashboardComponent to combine navbar and sidebar components
4. Update css to include styling for new components
5. Use jquery to toggle display of dashboard and signinComponents to reflect user interraction

nzediegwu1 committed:
Merge pull request #50 from nzediegwu1/ft/react-dashboard-component

Implement react dashboard component

Commit:
Implement addEvent react component

1. Create addEventComponent
2. Render addEvent component in dashboard component
3. Tweak css to correct bugs

nzediegwu1 committed:
Merge pull request #51 from nzediegwu1/ft/add-event-react-component

Implement addEvent react component

Commit:
Implement addCenter component

1. Create addCenterComponent.js from template
2. Create ModalHeader and ModalFooter components for reusability
3. Render AddCenter component in dashboard component

nzediegwu1 committed:
Merge pull request #52 from nzediegwu1/ft/add-center-react-component

Implement addCenter component

Commit:
Correct synthax errors in jsx code

Commit:
Implement myCenters component

1. Create myCenter component
2. Import myCenter component in dashboard
3. Implement react-router

nzediegwu1 committed:
Merge pull request #53 from nzediegwu1/ft/react-myCenter-component

Implement myCenters component

Commit:
Implement react-router in dashboard component

1. Use history API  to include routing event in sidebarComponent nav bar items

Commit:
Implement eventList component

1. Create eventList component using template
2. Implement dashboard route for eventList component

nzediegwu1 committed:
Merge pull request #54 from nzediegwu1/ft/event-list-react-component

Implement eventList component

Commit:
Create tableHead component to eliminate code repetition

Commit:
Implement ManageEvent component

1. Create manage event component
2. Create route for manageEvent component
3. Replace raw hostory api calls with createBrowserHistory api from reac

nzediegwu1 committed:
Merge pull request #55 from nzediegwu1/ft/manage-event-react-component

Implement ManageEvent component

Commit:
Implement centerDetails component

1. Create centerDetails component
2.  Create centerDetails route
3.  Create options component
4. Create facility component
5. Include ref in formgroup  parameter
6. Create ManageDetailsHeader component for reusability
7. Experiment form input value collection in signin component

nzediegwu1 committed:
Merge pull request #56 from nzediegwu1/ft/center-details-react-component

Implement centerDetails component

Commit:
Setup and experiment React-redux

1. Install redux and react-redux
2. Install babel plugin for object spread operator
3. Create store, reducer and action for redux experiment
4. Create testRedux component for experimenting react with redux

nzediegwu1 committed:
Merge pull request #57 from nzediegwu1/ft/center-details-react-component

Setup and experiment React-redux

Commit:
Consume signin API endpoint

1. Install axios node module
2. Make http post request in handleSubmit login event and return server responses
3. Save token into localStorage

nzediegwu1 committed:
Merge pull request #58 from nzediegwu1/ft/consume-signin-API-endpoint

Consume signin API endpoint

Commit:
Consume user signup API endpoint

1. Write logic to validate signup user input
2. Implement user login on signup success
3. Implement user logout to delete localStorage token and redirect to login page
4. Modify webpack  devServer settings to set historyApiFallback to true enable browser addressbar route calls

Commit:
fixed issue of reloading nested route from browser by adding a slash '/' in index.html script's src property

Commit:
Secure react components and consume get-all-events API endpoint

1. Install multer for file-upload and implement picture upload for events
2. Define populateEvents redux Actions method
3. Secure dashboard route using token
4. Use populateEvents action to render fetched events from API on dashboard component mount
5. Implement toggle rendering of 'Add Center' nav button based on logged in user account type
6. Implement function to redirect logged in user to dashboard on attempt to go to login page
7. Create reusable file for reusing code
8. Implement combineReducer in rootReducer

nzediegwu1 committed:
Merge pull request #59 from nzediegwu1/fx/secure-api-endpoints

Secure react components and consume get-all-events API endpoint

Commit:
Standardize react-router implementation

1. Implement react-router to use nested routing and dynamic paths by defining centerRouter and eventRouter components
2. Implement event action and reducer and binding to eventList
3. Implement functionality to redirect to homepage when token doesnt exist in localStorage
4. Implement function not to render AddCenter modal component if token.accountType is not admin
5. Implement setEventDetail action to be dispatched to store when rendering eventLists, to be used toget event details onClicking an event in the list
6. Implement logic to also check if logged in user id is same as creator of clicked event before rendering edit/delete icons in manageDetails component
7. ManageEvent component: onsume eventDetails from store on clicking an event

nzediegwu1 committed:
Merge pull request #60 from nzediegwu1/fx/standardize-react-router

Standardize react-router implementation

Commit:
Implement API endpoint for getting an event

1. Implement client function to get event and dispatch to redux store and render
2. Implement API endpoint for getting an event

nzediegwu1 committed:
Merge pull request #61 from nzediegwu1/ft/get-an-event-API-endpoint

Implement API endpoint for getting an event

Commit:
style index.html file

Commit:
Delete experimental redux implementation

Commit:
Consume addEvent API endpoint

1. Implement logout upon failed attempt to post new event, due to invalid or undefined token
2. Include Server/public into gitignore
3. Create changePage redux action to use track page location in other to either navigate to or refresh (via state)  eventDetails page upon successful event post
4. Delete and replace modal footer component for ease of use
5. Implement functionality to add an event from addEvent component
6. Integrate dashboard components using Routes for easy use of history props
7. Refractor logout function into reusables.js
8. Modify addEvent controller response to include associated user and center data

nzediegwu1 committed:
Merge pull request #62 from nzediegwu1/ft/consume-add-event-api-endpoint

Consume API endpoint for Adding an Event

Commit:
Consume API endpoint to retrieve single center and center details

1. Reorganize folder structure to eliminate redux folder
2. Move changePage action and reducer to new files
3. Create center action and reducer
4. Eliminate facility component and replace tables with dynamic table component
5. Resolve bug in manageDetails header implementation
6. Replace event details with table in manageEvent Component
7. Modify centerController to save picture uploaded
8. Modify validator to validate uploaded picture in center http requests
9. Integrate multer module in center router to enable file upload

Commit:
Consume API endpoint for adding a center

1. Remove unused mapStateToProps function used in addEventComponent
2. Modify add center controller action to return user and events on center creation
3. Consume the addCenter endpoint

nzediegwu1 committed:
Merge pull request #63 from nzediegwu1/ft/consume-add-center-API-endpoint

Consume API endpoint to retrieve single center, center details and add a center

Commit:
Consume API endpoint for deleting an event

1. Implement logic to send request to a delete a resource given pageType, param and picture props
2. Implement logic in controller delete action to delete image file from /public/events directory upon successful delete request

nzediegwu1 committed:
Merge pull request #64 from nzediegwu1/ft/consume-delete-event-API-endpoint

Consume API endpoint for deleting an event

Commit:
Consume API endpoint for updating an event

1. Implement functionality for switching center image field between required and not-required for creating or updating center respectively
2. Implement functionality to switch title of addEvent and addCenter modals between create and modify states
3.  Implement functionality to bind eventDetails and centerDetails state props to event and center form fields upon update action
4. create page actions for setting modal title, required field property, eventDefaults and centerDefaults and implement reducer handler
5. In manageDetailsHeader component, implement feature to dispatch setModalTitle, setRequired, setEventDefaults, and setCenterDefaults actions for either event or center update contexts
6. Prompt for validation of delete action in when user tries to delete an event/center
7.  In navBar component, implement feature to dispatch setModalTitle, setRequired, setEventDefaults, and setCenterDefaults actions for new event or center creation contexts
8. In event and center server Controllers: validate for picture field (center) and delete/relace picture file when an event/center image was updated in put http action
9. Remove validation for center picture in validator middleware

Commit:
1. Implement functionality to get centers and populate center selection input control in  add-event component

2. Make getCenter http request a reusable functionality

3. Call getCenter functionality inside addEvent componentWillMount method

Commit:
Consume API endpoint to update event and center

1. Implement functionality to send update request for a center
2.  Implement functionality to send update request for an event
3. Correct eventDefaults error in pageReducer.js
4. Implement functionality to eager load center events, facilities and user with update response when modifying a center
5.  Implement functionality to eager load event center, and user with update response when modifying an event
6. Correct error when modifying time of an event

Commit:
Delete template images which has been replaced by images from server

nzediegwu1 committed:
Merge pull request #65 from nzediegwu1/ft/consume-update-event-API-endpoint

Ft/consume update event api endpoint

Commit:
1. Include Template directory in eslintignore file
2. Reference eslintignore in .hound.yml file

Commit:
Refactor source code

1. Refactor verifyEvent method in validator.js

Commit:
Refactor source code

1.  Eliminate unnecessary build, postinstall and start:dev scripts
2. Setup eslint for react jsx
3. Format code using prettier code formatter
4. Refactor and format validator.js script

Commit:
Comment out tests temporarily

nzediegwu1 committed:
Merge pull request #66 from nzediegwu1/fx/refactor-source-code

Refactor validator source code

Commit:
1. Replace hand coded links with dynamic origin
2. Integrate environment plugin in webpack
3. Implement functionality to dynamically set api link depending on node_env
4. Include userid in signup api response

nzediegwu1 committed:
Merge pull request #67 from nzediegwu1/fx/refactor-source-code

Refactor source code for hosting to heroku

Commit:
Create a postinstall script

Commit:
modify webpack config to reflect latest changes

Commit:
switch postinstall and start scripts

Commit:
eliminate postinstall script

Commit:
1. Integrate history API fallback
2. Move babel plugin from devDependencies to dependencies
3. Replace react routing with BrowserRouter of react-router-dom

Commit:
remove client/dist/bundle.js from gitignore

Commit:
change app name to eventmanageronline

Commit:
change logic for process.env.NODE_ENV in client app

Commit:
updated react app bundle.js

Commit:
update apiLink in react app.js

Commit:
1. Implement reusable apiLink in reusables.js
2. Rebundle

Commit:
1. Fix eslint errors
2. Move apiLink localstorage variable assignment from app.js to routerComponent
3. Move index.html style into custom css class

Commit:
change apiLink to production

Commit:
Make correction in apiLink

Commit:
Refactor Server code to implement cloudinary

1. Eliminate bin/server.js and change app start script in package.json
2. Integrate app server in app.js
3. Integrate cloudinary file upload in center add and center modify api endpoints
4. Integrate cloudinary file upload and delete in event add, modify and delete api endpoints
5. Migrate event and center database tables to include public_id fields
6. Include public_id fields in sequelize event and center models
7. Refactor model and migration codes to obey eslint rules
8. Replace multer diskStorage with multer memoryStorage in event and center routers respectively

Commit:
1. Remove Server/public folder from gitignore and delete
2. Replace all image url in client to reflect change in image source
3. Recompile webpack bundle.js

nzediegwu1 committed:
Merge pull request #69 from nzediegwu1/fx/use-cloudinary-image-upload

Use cloudinary image upload

Commit:
changed error response of no-center/no-event to 404

Commit:
make changes to enable super user for first user signup

Commit:
correct error in rendering createCenter modal

Commit:
change position of apiLink declaration to eliminate initial undefined error

Commit:
correct apiLink used

Commit:
refractor apiLink position to resolve issue of 404 in getting centers

Commit:
Set apiLink variable from localStorage item to global reusable variable in reusable.js since it is not to be set or changed while application is running

Commit:
Implement functionality to manage center facilities

1.  Implement facility router and controller
2. Implement functionality to bulk create/destroy facilities in facility controller
3. Implement validator for facility bulk input
4. Modify facility model and migration to use UUID datatype in id property

Commit:
Changed error response for unauthorized transaction

nzediegwu1 committed:
Merge pull request #70 from nzediegwu1/ft/implement-add-facilities-api-endpoint

 Implement functionality to manage center facilities

Commit:
Restructure event/center posting to use client side image upload

1. Modify addCenter component to implement clientside image upload before server API call
2. Modify addEvent component to implement clientside image upload before server API call
3. Implement functionality to hide sidebar on navigation click
4. Modify center controller to not use cloudinary file upload in center adding and update
5. Modify event controller to not use cloudinary file upload in center adding and update
6. Use query to pass and obtain publicId of event upon deletion
7. Modify error response in eventController, centerController, validator and authenticator to first delete uploaded image from cloudinary
8. Change public_id field in even and center models to publicId
9. Eliminate multer implementation for file-upload in event and center router
10. Change event and center data format to be sent from client from formData to json

nzediegwu1 committed:
Merge pull request #71 from nzediegwu1/fx/use-cloudinary-client-side-image-upload

Reimplement event/center posting to use client side image upload

Commit:
Create component for managing facilities

1. Remove ui for managing facilities in addCenter component
2. Create manageFacilities component
3. Import manageFacilities component in centerDetails component and add a settings button for triggering the modal
4. Modify style for center details tables for better look

Commit:
Implement functionality to manipulate facilities on clientside

1. Create facility actions and reducer for data manipulation
2. Implement functionality to render facility from store in centerDetails
3. Implement functionality to add, delete and submit facilities in manageFacility component
4. Resolve eslint issues in reducers and actions
5. Add new style for managing facility

Commit:
Consume facilities api endpoint

1. Implement functinality to post data from manageFacility component to facilities api endpoint using axios

Commit:
Make corrections to eliminate errors in posting facilities

1. Implement functionality to close modal on successful facility request
2. Implement functionality to logout user when token is not authenticated when attempt is made to post facility
3. Correct unexisting key error in rendering facilities

Commit:
switch apiLink to production

nzediegwu1 committed:
Merge pull request #72 from nzediegwu1/ft/consume-facility-management-api-endpoint

Implement facility management in client side

Commit:
Prevent event creation in close center

1. Implemented conditional functionality to check availability of center upon event creation and modification
2. Return an error message if center is unavailable for event creation/modification

nzediegwu1 committed:
Merge pull request #73 from nzediegwu1/fx/prevent-event-creation-in-close-center

Prevent event creation in close center

Commit:
Define responseWithCloudinary reusable function in validator.js to replace event and center response messages for post and put requests, which eliminates app crashing

Commit:
Fix:

Implement functionality to delete old center/event image upon successful modification

Commit:
change folder names temporarily in other to rename according to convention

Commit:
remove unused script

Commit:
rename server, template and client folders according to convention

Commit:
correct server string in package.json test script

Commit:
Implement feature to enable an admin approve or reject an event

1. Create route for approving event
2. Include status property in event model and migration
3. Define approveEvent controller action to handle approval requests
4. Implement functionality to get only events that are approved in getAll route

nzediegwu1 committed:
Merge pull request #74 from nzediegwu1/ft/admin-can-approve-event

Implement feature to enable an admin approve or reject an event

Commit:
Consume API endpoint for approving event by center admin

1. Include bootstrap switch plugin in manageEvent page and modify  styles to fit
2. Implement functionality to approve or reject an event by center admin, by toggling status switch component
3. Modified approval API endpoint to return modified event in response, to be used in frontend databinding

Commit:
Implement feature to send email to user upon event approval/rejection

1. Installed nodemailer package
2. Send status email upon transaction success

nzediegwu1 committed:
Merge pull request #75 from nzediegwu1/ft/consume-api-endpoint-for-approving-events

 Consume API endpoint for approving event by center admin

Commit:
Refactor 3rd party API calls for speed
1. Refactor cloudinary image delete to be asynchronous and faster by moving end response from callbacks
2. Refactor email sending to be asynchronous and faster by moving end response from callbacks
3. Refactor validator responses to send response outside cloudinary callbacks

Commit:
Create template for profile page

1. Design image upload section
2. Design profile tab and tab-content
3. Design edit-profile tab and tab-content

nzediegwu1 committed:
Merge pull request #76 from nzediegwu1/ft/create-profile-template

Create template for profile page

Commit:
ignore DS_Store

Commit:
Create user profile component

1. Create profileComponent from template markup
2. Render profile component in dashboard component via react router
3. Add extra styles to beautify profile component

Commit:
modify css on profile nav tabs

nzediegwu1 committed:
Merge pull request #77 from nzediegwu1/ft/create-user-profile-component

Create user profile component

Commit:
Merge branch 'develop' of https://github.com/nzediegwu1/EventManager into develop

Commit:
Allow user post or update to rejected events date
1. Implement functionality in addEvent action
2. Implement functionality in modifgyEvent action
3. Implement functionality in approveEvent action

Commit:
fix bug in addEvent controller action

nzediegwu1 committed:
Merge pull request #78 from nzediegwu1/fx/allow-post-put-on-rejected-events-date

allow post put on rejected events date

Commit:
repackage bundle.js to reflect latest changes

Commit:
secure getUsers action in user controller to allow super admin alone

Commit:
1. Create route for fetching user profile
2. Implement controller action for fetching user profile
3. Refactor authenticator for speed

nzediegwu1 committed:
Merge pull request #79 from nzediegwu1/ft/API-endpoint-for-getting-user-profile

API endpoint for fetching user profile

Commit:
Implement API endpoint for modifying user profile

1. Implement controller action to modify user profile
2. Create route for user profile update
3. Implement functionality to prevent update to an existing username, phone number or email

Commit:
Update migration and model files to include extra profile details

nzediegwu1 committed:
Merge pull request #80 from nzediegwu1/ft/API-endpoint-to-modify-user-profile

Implement API endpoint for modifying user profile

Commit:
Implement API endpoint to handle account upgrade

1. Create route for account upgrade
2. Define user controller action to handle account accout upgrade
3. Implement functionality to prevent non-super user from performing action
4. Implement functionality to prevent super user from ugrading own account
5. Implement functionality to validate acccount type before performing db transaction

Commit:
change user upgrade route to convention

nzediegwu1 committed:
Merge pull request #81 from nzediegwu1/ft/API-endpoint-for-account-upgrade

Implement API endpoint to handle account upgrade

Commit:
fix minor validation issues in upgrade user action

Commit:
fix error in preventing super user account upgrade

Commit:
Implement API endpoint to change profile image

1. Create route to allow user change profile image
2. Include profile picture and public id fields in model and migrations
3. Implement validation for profile image upload request
4. Implement controller action to allow user change profile picture

nzediegwu1 committed:
Merge pull request #83 from nzediegwu1/ft/API-endpoint-to-change-profile-picture

Implement API endpoint to change profile image

Commit:
1. Fix error in calling validator function for user signin
2. Create redux userAction for setting profile details
3. Refactor profile component for reusability of form inputs
4. Include profileData state in userReducer

Commit:
Consume API endpoint for getting profile details

1. Implement functionality to retrieve profile details via axios and dispatch to store on component mount
2. Implement functionality to render profile details from store
3. Use reusable table rows in profile component
4. Make route for getting profile details unsecure

nzediegwu1 committed:
Merge pull request #84 from nzediegwu1/ft/consume-API-endpoint-for-profile-details

Consume API endpoint for getting profile details

Commit:
Consume API endpoint for updating profile

1. Implement functionality to send http put request for updating user profile
2. Implement functionality to dispatch response after update request
3. Implement functionality to reset input controls

nzediegwu1 committed:
Merge pull request #85 from nzediegwu1/ft/consume-api-endpoint-for-updating-profile

Consume API endpoint for updating profile

Commit:
Fx: set required property of input controls

Commit:
Recompile bundle,js

Commit:
Consume API endpoint for account upgrade

1. Implement functionality to selectively render control for upgrading an account by super admin

Commit:
Implement Feature to getUserList

1. Define populateUserList action and modify userReducer to correspond
2. Create userList component
3. Implement functionality to conditionally render userList component if user is super admin or not
4. Consme userList API endpoint

Commit:
Implement functionality to allow super admin to modify account type

nzediegwu1 committed:
Merge pull request #86 from nzediegwu1/ft/consume-API-endpoint-for-upgrading-account

Consume api endpoint for account upgrade

Commit:
Implement feature to enable user upload profile picture

1. Implement functionality to upload picture to cloudinary in client
2. Implement functionality to post picture details to server for update

nzediegwu1 committed:
Merge pull request #87 from nzediegwu1/ft/consume-api-endpoint-for-profile-picture

Implement feature to enable user upload profile picture

Commit:
Rebuild bundle.js to reflect latest changes in client app

megwali commented on .sequelizerc:
```diff
@@ -0,0 +1,8 @@
+	"models-path": path.resolve('./Server/models'),
+	"seeders-path": path.resolve('./Server/seeders'),
+	"migrations-path": path.resolve('./Server/migrations'),
+};
```
Is there a reason why you are using `Server` instead of `server` since your actual folder says `server`?

nzediegwu1 replied:
I refactored my code folder from Server to sever, would reflect the changes asap

megwali commented on README.md:
```diff
@@ -1,2 +1,190 @@
+
 Given you manage an events center, this app will help you accept applications to use your center/facilities, and will either decline events when the proposed day is already taken, or suggest an available day
+
+###### App's Documentation: https://eventmanager29.herokuapp.com/docs
```
Link does not work. Please update or remove.

nzediegwu1 replied:
Will make corrections. Thanks

megwali commented on README.md:
```diff
@@ -1,2 +1,190 @@
+
+
+## Setup guide: Clone this repository. To install all dependencies, run;
+
```
NITPICK: How do I clone the repo :)
Add git clone {URL}

nzediegwu1 replied:
Thanks for the suggestion. Will do

megwali commented on README.md:
````diff
@@ -1,2 +1,190 @@
+
+
+
+``` npm run start:dev  ```
````
This command doesn't work (anymore)? Update to match what you have in your package.json

nzediegwu1 replied:
Thanks a lot, will update

megwali commented on README.md:
````diff
@@ -1,2 +1,190 @@
+
+
+
+``` npm run start:dev  ```
````
This command doesn't work (anymore)? Update to match what you have in your package.json

nzediegwu1 replied:
Will modify. Thanks for the feedback

megwali commented on README.md:
```diff
@@ -1,2 +1,190 @@
+  </tr>  
+  <tr>
+      <td>PUT</td>
+      <td>/api/v1/events/<eventId> </td>
```
Use `/api/v1/events/:eventId` 1. It is the convention 2. Markdown on Github sees anything in `<>` as a tag and doesn't display it.

nzediegwu1 replied:
Thanks for the feedback, will do

megwali commented on README.md:
```diff
@@ -1,2 +1,190 @@
+  
+  <tr>
+      <td>DELETE</td>
+      <td>/api/v1/events/<eventId></td>
```
Update all instances as per the last comment.

nzediegwu1 replied:
Will do that as soon as possible

megwali commented on README.md:
```diff
@@ -1,2 +1,190 @@
+      <td>GET</td>
+      <td>/api/v1/events/</td>
+      <td>Get all events</td>
+      <td></td>
```
No sample payload?

nzediegwu1 replied:
Not at all.

megwali replied:
A sample payload that tells me what to expect from the api would be useful.

megwali commented on .gitignore:
```diff
@@ -4,3 +4,16 @@
+/.vs
+/node_modules
+/server/typings
+/Server/dist
```
Add client/dist to this list

nzediegwu1 replied:
My server actually serves the index.html + bundle.js from that folder, removing it would make the client inaccessible. Is there a better way to render my react app from node.js server?

megwali replied:
Try html-webpack-plugin.
These links should be helpful: https://webpack.js.org/plugins/html-webpack-plugin/
https://www.codecademy.com/articles/react-setup-iv

Ensure dist is not pushed to github.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+
+const mapDispatchToProps = dispatch => ({
+  setCenterDetails: center => dispatch(setCenterDetails(center)),
+  // setCenterDefaults: data => dispatch(setCenterDefaults(data)),
```
Remove all commented code.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+import addressIcon from '../resources/images/glyphicons-243-map-marker.png';
+import cityIcon from '../resources/images/glyphicons-740-cadastral-map.png';
+import capacityIcon from '../resources/images/glyphicons-44-group.png';
+import amountIcon from '../resources/images/glyphicons-228-usd.png';
```
You could create an `<Icon>` component that simply takes a prop of the image/icon name and returns the Icon. This allows you to import only `Icon` and use where necessary rather than import 6 or more individual icons.

nzediegwu1 replied:
Thanks for the brilliant suggestion. Will it be better to host all the images on cloudinary and simply enter the link in image src such that the size of bundle.js file would be reduced

megwali replied:
That decision is left for you. The `pro` like you pointed out is that you outsource the job to (hopefully) a more reliable platform and have a smaller bundle. The `con` is that you yield control and have to make additional external api calls for your app to work properly.

andela-megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+import addressIcon from '../resources/images/glyphicons-243-map-marker.png';
+import cityIcon from '../resources/images/glyphicons-740-cadastral-map.png';
+import capacityIcon from '../resources/images/glyphicons-44-group.png';
+import amountIcon from '../resources/images/glyphicons-228-usd.png';
```
You could create an `<Icon>` component that simply takes a prop of the image/icon name and returns the Icon. This allows you to import only `Icon` and use where necessary rather than import 6 or more individual icons.

andela-megwali replied:
You could create an `<Icon>` component that simply takes a prop of the image/icon name and returns the Icon. This allows you to import only `Icon` and use where necessary rather than import 6 or more individual icons.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+    const capacity = this.capacity.value;
+    const price = this.price.value;
+    const availability = this.availability.value;
+    const token = JSON.parse(localStorage.token).value;
```
I don't think these constants above are necessary. Most of them are only used once after the definition.

nzediegwu1 replied:
Thanks, will refactor asap

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+    const modalTitle = this.props.modalTitle;
+    const history = this.props.history;
+    const setCenter = this.props.setCenterDetails;
+
```
Use JS Object destructuring for props.

nzediegwu1 replied:
Thanks for the feedback, will do

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.folder = apiLink === 'http://localhost:8080' ? 'dev/centers' : 'prod/centers';
+  }
+  handleSubmit(event) {
```
CRITICAL: I think this method is doing too much and is too long. You should consider breaking out smaller functions and calling those functions within this one.

NITPICK: You might also want to consider moving some of the actions / api logic into an `actions` file and import/call the functions e.g. `createCenter`, `updateCenter` here.

CRITICAL: Remove all `alert` calls and replace with a more user friendly notification like a `toast`.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+    } else {
+      saveCenter(undefined);
+    }
+  }
```
NITPICK: Please leave a space between functions/methods. It helps with readability.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+    }
+  }
+  componentWillReceiveProps(nextState) {
+    const centerDefaults = nextState.centerDefaults;
```
use object destructuring

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+  required: state.page.required,
+  centerDefaults: state.page.centerDefaults,
+});
+let centerId;
```
I don't think this is necessary.

andela-megwali replied:
I don't think this is necessary.

andela-megwali replied:
I don't think this is necessary.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+    centerId = centerDefaults.id;
+  }
+  render() {
+    const content = (
```
This is not necessary, just return your `jsx`

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+import React, { Component } from 'react';
```
Most of the feedback for `/addCenterComponent.js` applies here too. I also see they use similar functions. You should consider extracting these functions and calling them where needed to keep things DRY.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+          history.push(`/dashboard/centers/${response.data.data.id}`);
+          setCenter(response.data.data);
+          if (modalTitle !== 'New Center') {
+            $('#addNewCenter').modal('hide');
```
Weird to see jquery in a react app. Instead pass a `show` or `visibility` prop to your modal that determines if it is hidden or not.

megwali commented on client/src/components/addCenterComponent.js:
```diff
@@ -0,0 +1,234 @@
+  render() {
+    const content = (
+      <div
+        className="modal fade"
```
Your `visibility` prop should be placed somewhere around here.

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+          }
+        })
+        .catch(err => {
+          typeof err.response.data.message !== 'object' &&
```
This should be in an if block.

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+            alert(JSON.stringify(err.response.data.message));
+          (err.response.status === 403 || err.response.status === 401) &&
+            logout('addNewEvent', history);
+          if (typeof err.response.data.message === 'object') {
```
Assuming the logout above is in an if block and returns or redirects the user to another page, you would not need the if block here.

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+    hour = hour < 10 ? `0${hour}` : hour;
+    // It adds a 0 to number less than 10 because input[type=time] only accepts 00:00 format.
+    min = min < 10 ? `0${min}` : min;
+    const eventDate = `${year}-${month}-${day}`;
```
You could evaluate line 129 in this block and use a const on line 125 instead.

andela-megwali replied:
You could evaluate line 129 in this block and use a const on line 125 instead.

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+    let month = date.getMonth() + 1; // This method gets the month and Jan is 0.
+    let day = date.getDate(); // This method gets the day of month as a number.
+    let hour = date.getHours(); // This method gets the hour
+    let min = date.getMinutes(); // This method gets the minutes
```
Preferably, these should be `const`

nzediegwu1 committed:
Update readme to reflect mentors feedback

Commit:
1. Replace .eslintre file with .eslintre.json file to resolve hound error
2. Spelling correction in .sequelizerc file
3. Eliminate unused code in .gitignore

Commit:
Merge branch 'develop' of https://github.com/nzediegwu1/EventManager into develop

Commit:
refactor addCenterComponent.js in line with mentor feedback

Commit:
modify .eslint file to .eslint.json

Commit:
removed comment in .eslint file

nzediegwu1 committed:
Merge branch 'develop' into fx/mentor-feedback

Commit:
remove react plugin import in extends

Commit:
Merge branch 'fx/mentor-feedback' of https://github.com/nzediegwu1/EventManager into fx/mentor-feedback

nzediegwu1 committed:
Merge pull request #88 from nzediegwu1/fx/mentor-feedback

refactor addCenterComponent.js in line with mentor feedback

megwali commented:
General Feedback: Your test coverage needs to go up to 70% or more (actually there really is no test right now as all of them are commented out when I checked). Also can you drop the term `mentor`?

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+              occupiedDates += `${new Date(date).toDateString()}\n`;
+            });
+            alert(
+              `MESSAGE:\n${err.response.data.message.Sorry}\n\nOCCUPIED DATES:\n${occupiedDates}`
```
String literals do not need `\n` for a new line. Just write your next line on the next line. It makes it more readable too.

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+            $('#addNewEvent').modal('hide');
+          }
+        })
+        .catch(err => {
```
create a const that extracts `status` and `message` since they are used severally and are nested deep in the object.

megwali commented on client/src/components/addEventComponent.js:
```diff
@@ -0,0 +1,250 @@
+          (err.response.status === 403 || err.response.status === 401) &&
+            logout('addNewEvent', history);
+          if (typeof err.response.data.message === 'object') {
+            let occupiedDates = '';
```
NITPICK: Preferably use a `<p>` tag and do:
message.OccupiedDates.forEach(date => {
   `<p>`new Date(date).toDateString()`</p>`;
 });

megwali commented on template/pages/profile.html:
```diff
@@ -0,0 +1,207 @@
+              <div class="table-responsive">
+                <table class="table table-hover table-striped table-bordered">
+                  <tbody>
+                    <tr>
```
Why do you have hardcoded user info?

megwali commented on template/pages/Signup.html:
```diff
@@ -0,0 +1,169 @@
+                        <span class="input-group-addon">
+                            <img src="../resources/images/glyphicons-11-envelope.png" alt="email" />
+                        </span>
+                        <input type="email" name="email" placeholder="Email" required class="form-control input-lg" />
```
You should implement validation (front and back end) for your forms

megwali commented on template/pages/MyEvents.html:
```diff
@@ -0,0 +1,549 @@
+    <script src="../resources/scripts/jquery.js"></script>
+    <script src="../resources/scripts/popper.min.js"></script>
+    <script src="../resources/scripts/bootstrap.js"></script>
+    <style>
```
Can we create a separate file for the styles (in all affected areas) rather than mixing it with the html.

Commit:
add margin space to hound file lines

nzediegwu1 committed:
fix error with hound settings spacing

Commit:
remove extra spacing in .hound.yml file

nzediegwu1 committed:
Merge branch 'develop' into fx/mentor-feedback

nzediegwu1 committed:
Merge pull request #89 from nzediegwu1/fx/mentor-feedback

Implement feedback

Commit:
[#157219929] Refactor client source code

1. Created services.js to handle all http transactions.
2. Move reusables.js into service.js and deleted reusables.js
3. Make client/src/resources/images static in app.js for easy reference in jsx image srcs
4. Refactor addCenterComponent.js to eliminate images imports
5. Refactor addCenterComponent.js to handle http requests in services.js
6. Refactor addEventComponent.js to eliminate images imports
7. Refactor addEventComponent.js to handle http requests in services.js
8. Create Icon component to render icons images
9. Refactor centerDetails component to use Icon component and handle http requests in services.js
10. Refactor centerList component.js to handle http requests in services.js
10. Refactor eventList component.js to handle http requests in services.js
11. Refactor formGroup component to allow use of image name directly in src rather than imported image
12. Refactor manageDetailsHeader component to use Icon component and handle http requests in services.js
13. Refactor manageEvent component to use Icon component and handle http requests in services.js
14. Refactor manageFacilities component to use Icon component and handle http requests in services.js
15. Refactor navbar component to use Icon component
16. Refactor profile component to handle http requests in services.js
17. Refactor sidebar and recoverPassword components to use Icon component
18. Refactor signin and signup components to use Icon component and handle http requests in services.js
19. Refactor userList component to handle http requests in services.js

Commit:
[#157219929] Resolve .hound.yml invalid format

nzediegwu1 committed:
Merge branch 'develop' into fx/code-refactoring-157219929

Commit:
[#157219929] Resolve invalid houndci format

nzediegwu1 committed:
Merge pull request #90 from nzediegwu1/fx/code-refactoring-157219929

#157219929 Refactor client source code

Commit:
[Fix #157370148] Refactor frontend and improve app User Experience

1. Integrate font-awesome to handle submit processing feedback
2. Replace alerts with toastr
3. Discard component validate method and use service userValidator
4. Refactor table.js component implementation to use .map method, use random key and eliminate colNumber prop
5. Remove unnecessary content variables in component render methods
6. Validate profile data onSubmit
7. Add object-fit css in .profile-image class
[Finishes #157370148]

Commit:
Update bundle.js file with changes made

nzediegwu1 committed:
Merge pull request #91 from nzediegwu1/fx-improve-app-UX-157370148

[Fix #157370148] Refactor frontend and improve app User Experience

Commit:
Update bundle.js with latest build

Commit:
Correct userList table column naming

Commit:
[Feature #157533803] Secure Facility Component and refactoring

1. Implement functionality to relectively render Facility component if current user is center owner
2. Refactor client code affected as follows:
   a. Include dist client and server folders to gitignore
   b. Include cloudinary API_KEY, UPLOAD_PRESET and API_SECRET in .env.sample file
   c. Install webpack modules for minimizing app bundle size
  d. Change start script to be optimized for production in package.json
  e. Include build and postinstall scripts in package.json
  f. Create separate webpack configs for dev and production which takes from webpack.common.config
  g. Comment out wrongly swagger setup in server/app.js
[Finishes #157533803]

nzediegwu1 committed:
Merge pull request #92 from nzediegwu1/fx/secure-facility-component-157533803

#157533803 Secure Facility Component and refactoring

Commit:
Delete files in dist folder

Commit:
refactor .gitignore dist ignore settings

Commit:
Fix undefined error with environment plugin

Commit:
Correct undefined environment variable error

Commit:
[Feature #157672601] Implement password recovery

1. Define recoverPassword method in services.js
2. Define handler for sending http request to recover password onSubmit of recovery form
3. Create route to serve password recovery requests in server
4. Define recoverPassword controller action to process password recovery and send email
[Finishes #157672601]

nzediegwu1 committed:
Merge pull request #93 from nzediegwu1/ft/implement-password-recovery-157672601

#157672601 Implement password recovery

Commit:
Move modal closing action to success event

Commit:
Add request processing feedback onSubmit in password recovery functionality

Commit:
[Fix #157693752] Fix inconsistent imnplementations

1. Conditionally render events that were rejected on red color with a (rejected) text appended
2. Conditionally render center availability on red color when closed
3. In centerList page, conditionally render closed centers on red
[Finishes #157693752]

nzediegwu1 committed:
Merge pull request #94 from nzediegwu1/fx/inconsistent-implementations-157693752

#157693752 Fix inconsistent imnplementations

Commit:
[Chore #157720813]

Implement PropType validation of component props

[Starts #157720813]

Commit:
[Chore #157720813] Add Comments to client code

1. Add JsDoc comment to methods in services.js
2. Add extra descriptive comments to client source code

[Finishes #157720813]

nzediegwu1 committed:
Merge pull request #95 from nzediegwu1/ch/enforce-code-standard-157720813

#157720813 Enforce code quality standards

Commit:
[Chore #157720813] Enforce code quality standards

1. Move email credentials into environment variables
2. Install node-sass and sass-loader
3. Convert custom.css to custom.scss

[Finishes #157720813]

Commit:
[Feature #157733863] Implement pagination

1. Implement server side pagination for event list
2. Implement server side pagination for center list
3. Implement server side pagination for user list

[Starts #157733863]

Commit:
[Feature #157733863] Implement client side pagination

1. Install react-js-pagination, babel-presets-es2015 and stage-0 dependencies
2. Implement functionality to handle pagination in userList component
3. Implement functionality to handle pagination in centerList component
4. Implement functionality to handle pagination in eventList component
5. Implement functionality to handle pagination in centerList of addEvent component
6. Define redux action and constant getEventCenters and GET_EVENT_CENTERS respectively
7. Define redux action and constant setDataCount and SET_DATA_COUNT respectively
8. Set pagination LIMIT constant globally
9. Implement functionality to retrieve center selected in addEvent component
10. Handle SET_DATA_COUNT and GET_EVENT_CENTERS in page and center reducers
11. Add scss styling for pagination feature
12. Modify API json response for easy integration with frontend pagination feature

nzediegwu1 committed:
Merge pull request #96 from nzediegwu1/ft/implement-pagination-157733863

#157733863 Implement pagination

Commit:
Specify node version to use in production

Commit:
Install transform-react-remove-prop-types babel plugin to fix issue with pagination plugin on heroku

Commit:
Merge pull request #96 from nzediegwu1/ft/implement-pagination-157733863

#157733863 Implement pagination

Commit:
Merge branch 'develop' of https://github.com/nzediegwu1/EventManager into develop

Commit:
Modify style of pagination component in addEvent component

Commit:
[Feature #152860107] Implement frontEnd search

1. Define function to handle table search
2. Modularize ModalFooter react component
3. Modularize Filter search component
4. Remove search form in navBar component

nzediegwu1 committed:
Merge pull request #97 from nzediegwu1/ft/implement-search-feature-152860107

#152860107 Implement front-end search

Commit:
Remove engine property in package.json to speedup deployment time

Commit:
Merge branch 'develop' of https://github.com/nzediegwu1/EventManager into develop

Commit:
Include code-climate config and exclude template folder

Commit:
Delete content of template folder and replace with link to gh-pages branch

Commit:
Modify REAMDE.md file for aestatics

Commit:
Modify Readme.md

Commit:
[Fix #157219929] Refactor backend source code

1. Create util.js file in server root to house reusable functions
2. Move reusable response functions from validator to util.js
3. Move confimParams function to util from validator.js
4. Move email and cloudinary configs into util.js
5. Restructure validator.js to be more consize, and use best practices

[Starts #157219929]

Commit:
[Fix #157219929] Refactor serverside code

1. Define compareCenters function to handle center check when creating or updating a center
2. Define centerEntry, eventEntry and UserEntry which takes request and returns data object for db transaction
3. Define checkAvailability function to handle checking for center availability when creating or updating an event
4. Create services.js to handle reoccuring database transaction types: findById, getAll, updateImmediate, update

[Finishes #157219929]

nzediegwu1 committed:
Merge pull request #98 from nzediegwu1/fx/refactor-server-sourcecode-157219929

#157219929 Refactor backend source code

Commit:
Delete error causing codeclimate.yml file in cc build

Commit:
Add .codeclimate.yml file

Commit:
[Fix #157219929] Refactor backend source code

1. Exclude server/migrations in code climate config
2. Modify logout function to better reusability in client/service.js
3. Define updateAndEmail, create and modifyEvent functions in server/service.js for better reusability
4. Validate event approval route

[Finishes #157219929]

Commit:
Remove isEventType method in validator.js due to codeclimate violations

nzediegwu1 committed:
Merge pull request #99 from nzediegwu1/fx/refactor-server-code

#157219929 Refactor backend source code

Commit:
unexclude code climate path: client

Commit:
[Fix #157219929] Refactor client code and fix bugs

1. Increase toastr timeout from 2500 to 3500 for better UI
2. Modify getOne function in services to handle eventCenter type
3. Include setting for redux devtool in store
4. Create pageAction: setActivePge to replace component states: activePages
5. Create pageAction: SetRandom to handle error in binding event data to event form while modifying event
6. Create submitAction: setSubmitState to replace form component states: submitState
7. Replace eventCenter table with responsive grid view
8. Modify manageDetailsHeader.js to fix multiple errors in eventModal states
9. Convert class methods to ES6 arrow functions and eliminate unused constructors
10. Fix 404 error using pagination for gettingAll events due to unused query condition in findAndCountAll function

[Finishes  #157219929]

nzediegwu1 committed:
Merge pull request #100 from nzediegwu1/fx/refactor-client-code-157219929

#157219929 Refactor client code and fix bugs

Commit:
Hide center select control in addEvent modal

Commit:
Bug Fixes in react app

Add center Modal:
1. Implement functionality to wipe picture on modal close
2. Clear "undefined" text returned when component receives initial null state as props

AddEvent Modal
3. Implement functionality to wipe picture on modal close
4. Clear "undefined" text returned when component receives initial null state as props
5. Implement handleChange function to tackle problem of wiping input form when handlePageChange occurs

CenterDetails component
6. Modify centerOwner variable definition to solve issue of not rendering owners management icons after state change due to new center added

Commit:
Fix css issue in responsive table to allow overflow for very long texts

Commit:
[Fix #158759549] Implement server side feedback from Learning Facilitator

1. Remove password field in signup json response
2. When a regular user tries to update a center, validate account type first, before checking for already existing center
3. Limit array of events returned when retrieving a single center
4. Replace all 406 (Not acceptable) error response status codes to more suitable 409
5. Resolve event update error (request was not returning a response after previous feedback was implemented)
6. Error response messages should be more descriptive

[Finishes #158759549]

nzediegwu1 committed:
Merge pull request #101 from nzediegwu1/fx-implement-feedback-158759549

#158759549 Implement server side feedback from LF

Commit:
[Chore #152913098] Write API tests

1. Create migrations script in package.json to reset test database
2. Modify test script to run migration script before tests
3. Write tests for `users` endpoint
4. Write tests for `centers` endpoint
5. Write tests for `events` endpoint
6. Update .travis.yml file to contain other environment variables needed to run app
7. Fix status code error in findById service method to reflect various contexts
8. Fix signup error discovered from tests
9. Ignore `server/tests` folder in .codeclimate.yml settings

[Starts #152913098]

Commit:
[Chore #152913098] Write API tests

1. Write tests for facility endpoint

[Continues #152913098]

nzediegwu1 committed:
Merge pull request #102 from nzediegwu1/ch-write-api-tests-152913098

#152913098 Write API tests

Commit:
Remove environment variables from .travis.yml file

Commit:
Setup for code-climate coverage reporting in .travis.yml

nzediegwu1 committed:
Add code-climate test badge to readme.md

nzediegwu1 committed:
Update coveralls badge with html tag

nzediegwu1 committed:
Update coveralls badge with markdown

nzediegwu1 committed:
Update readme to reflect latest changes

nzediegwu1 committed:
Modify readme to reflect features

nzediegwu1 committed:
Update readme to be more descriptive

nzediegwu1 committed:
Correct mackdown error in readme

nzediegwu1 committed:
Update readme indentations

nzediegwu1 committed:
Correct spacing error in readme

nzediegwu1 committed:
Update readme to remove 'available taskrunners'

nzediegwu1 committed:
Modify readme "how to contribute"

nzediegwu1 committed:
Replace asterisks with numbering in readme

Commit:
Random fixes

Commit:
Merge branch 'develop' of https://github.com/nzediegwu1/EventManager into develop

Commit:
Updage readme to reflect new swaggar doc path

Commit:
Update licence and copyright in readme

Commit:
[Chore swaggar Doc] Update swaggar documentation

1. Update swaggar documentation to reflect latest state of API
2. Cleanup non-standard codes

[Finishes Swaggar Doc]

nzediegwu1 committed:
Merge pull request #103 from nzediegwu1/ch-swaggar-documentation

[Chore swaggar Doc] Update swaggar documentation
